### PR TITLE
Kan 39 create decks api

### PR DESCRIPTION
**Pull request by doroshkent**

### CODE REVIEW

File: src/pages/decksPage/decksApi.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        },
+      }),
+      getDecks: builder.query<DecksListResponse, GetDecksArgs | void>({
+        query: args => {
+          return {
+            method: 'GET',
+            params: args ?? undefined,
+            url: 'v2/decks',
+          }
+        },
```
Comment (stasfilippov) [inline src/pages/decksPage/decksApi.ts:31]: for query add - providesTag: ['nameTag']

File: src/pages/decksPage/decksApi.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  endpoints: builder => {
+    return {
+      createDeck: builder.mutation<CreateUpdateDeckResponse, CreateDeckArgs>({
+        query: args => {
+          return {
+            body: args,
+            method: 'POST',
+            url: 'v1/decks',
+          }
+        },
```
Comment (stasfilippov) [inline src/pages/decksPage/decksApi.ts:22]: add - invalidatesTags: ['nameTag']

File: src/pages/decksPage/decksApi.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      }),
+      updateDeck: builder.mutation<CreateUpdateDeckResponse, UpdateDeckArgs>({
+        query: ({ id, ...body }) => {
+          return {
+            body,
+            method: 'PATCH',
+            url: `v1/decks/${id}`,
+          }
+        },
```
Comment (stasfilippov) [inline src/pages/decksPage/decksApi.ts:48]: add - invalidatesTags: ['nameTag']

File: src/pages/decksPage/decksApi.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      }),
+      removeDeck: builder.mutation<void, RemoveDeckArgs>({
+        query: ({ id }) => {
+          return {
+            method: 'DELETE',
+            url: `v1/decks/${id}`,
+          }
+        },
```
Comment (stasfilippov) [inline src/pages/decksPage/decksApi.ts:39]: add - invalidatesTags: ['nameTag']

**Review by VladZimin — APPROVED** [review]

**Pull request merged**